src/main/java/com/commercetools/sync/commons/helpers/BaseSyncStatistics.java

```diff
@@ -202,6 +225,8 @@ public String getLatestBatchHumanReadableProcessingTime() {
     /**
      * Gets the number of days it took to process.
      *
+     * <p>Note: This method isn't thread-safe and shouldn't be used in a concurrent context.
```

Comment:
Why this method is not thread-safe?

Reply:
ah, @heshamMassoud Lam is right, the simple getters are thread-safe in this case:
they return primitives (or reference in case of String), so it's safe to call them.

Reply:
@lojzatran @andrii-kovalenko-ct ok,
1. but wouldn't the getter return not a correct value if called concurrently?
2. @andrii-kovalenko-ct Also, what do you mean "primitives -> so safe", so if the getter returns a `collection`/object it's not safe anymore?

Reply:
>but wouldn't the getter return not a correct value if called concurrently?

that's why you should decide what do you mean at "correct/not correct" values? Sequential calls of this method during batch processing could return different values, but they still all are correct in time they are called.

> Also, what do you mean "primitives -> so safe"

I was a bit wrong: `int` and `String` values are safe, because they are updated atomically, `long` (theoretically) won't be safe because it is multiword thus might require more than one tick to update entire value:
https://stackoverflow.com/a/9278798/907576

Reply:
Now i see the sync statistic became confusing with all this thread-specific staff.

Guess (might be in another task/PR) we have to use some synchronized builder and return immutable object which has some finished state, e.g. no one can call the statistic before it is finished completely, so no one gets uncompleted state of an update.

Reply:
What I mean by not "correct" is that it might have been already set by another thread, and the current thread doesn't yet see this updated value. That's why I think the comment makes sense to leave it. It's not thread safe because that getter does not ensure that a thread will see the latest value, as the value may be stale. 

> Guess (might be in another task/PR) we have to use some synchronized builder and return immutable object which has some finished state, e.g. no one can call the statistic before it is finished completely, so no one gets uncompleted state of an update.

Yep, I agree. I linked it to another issue. So it can be also fixed in the statistics refactoring. #157

Reply:
:+1